slack-api-client/src/test/java/test_with_remote_apis/scim/ApiTest.java

```diff
@@ -165,6 +186,78 @@ public void createAndDeleteUser() throws Exception {
+
+            UsersReadResponse readResp = slack.scim(orgAdminToken).readUser(req -> req.id(creation.getId()));
+            assertThat(readResp.getId(), is(creation.getId()));
+            assertThat(readResp.getUserName(), is(creation.getUserName()));
+            assertThat(readResp.getSlackGuest().getType(), is(User.SlackGuest.Types.MULTI));
+            assertThat(readResp.getSlackGuest().getExpiration(), is(notNullValue()));
+
+            try {
+                // https://api.slack.com/scim#filter
+                // filter query matching the created data
+                String userName = creation.getUserName();
+                UsersSearchResponse searchResp = slack.scim(orgAdminToken).searchUsers(req -> req.count(1).filter("userName eq \"" + userName + "\""));
+                assertThat(searchResp.getItemsPerPage(), is(1));
+                assertThat(searchResp.getResources().size(), is(1));
+                assertThat(searchResp.getResources().get(0).getId(), is(creation.getId()));
+                assertThat(searchResp.getResources().get(0).getUserName(), is(creation.getUserName()));
+                assertThat(searchResp.getResources().get(0).getSlackGuest(), is(notNullValue()));
+
+                String originalUserName = creation.getUserName();
+
+                User user = new User();
+                user.setUserName(originalUserName + "_ed");
+                slack.scim(orgAdminToken).patchUser(req -> req.id(creation.getId()).user(user));
+
+                readResp = slack.scim(orgAdminToken).readUser(req -> req.id(creation.getId()));
+                assertThat(readResp.getId(), is(creation.getId()));
+                assertThat(readResp.getUserName(), is(originalUserName + "_ed"));
+
+                User modifiedUser = readResp;
+                modifiedUser.setUserName(originalUserName + "_rv");
+                modifiedUser.setNickName(originalUserName + "_rv"); // required
+                slack.scim(orgAdminToken).updateUser(req -> req.id(modifiedUser.getId()).user(modifiedUser));
+
+                readResp = slack.scim(orgAdminToken).readUser(req -> req.id(modifiedUser.getId()));
+                assertThat(readResp.getId(), is(creation.getId()));
+                assertThat(readResp.getUserName(), is(originalUserName + "_rv"));
+                assertThat(readResp.getSlackGuest().getType(), is(User.SlackGuest.Types.MULTI));
+                assertThat(readResp.getSlackGuest().getExpiration(), is(notNullValue()));
+
+            } finally {
+                UsersDeleteResponse deletion = slack.scim(orgAdminToken).deleteUser(req -> req.id(creation.getId()));
+                assertThat(deletion, is(nullValue()));
+
+                // can call twice
+                UsersDeleteResponse deletion2 = slack.scim(orgAdminToken).deleteUser(req -> req.id(creation.getId()));
+                assertThat(deletion2, is(nullValue()));
+
+                // Verify if the deletion is completed (there is some delay)
```

Comment:
As the actual deletion can be done asynchronously after the DELETE endpoint call, we check if the deletion is completed by periodically calling the read API.

Reply:
What is the purpose of issuing the delete user call twice? And then a third time after the deletion is confirmed?

Reply:
@filmaj 
>What is the purpose of issuing the delete user call twice? And then a third time after the deletion is confirmed?

These code are basically taken from the existing one that creates Slack full members. The intention here is just to make sure if the current behavior is not changed. Currently, the server-side accepts deletion requests even after the user is already deleted. 

The assertion here is not for the quality assurance of the SDK itself. I added tests like this for the purpose to detect the server-side changes easily and quickly (I have been reporting detected issues to the server-side teams, thanks to those tests).

When I implemented the SCIM client, I felt that the deletion behavior is a bit surprising. Thus, I wanted to easily know if the behavior is changed in the future (I hope it won't happen though). If the behavior is changed in the future, I will communicate with the server side teams as necessary.

Reply:
OK, good to know, thanks for explaining